src/test/java/lv/ctco/javaschool/goal/boundary/GoalApiTest.java

```diff
@@ -281,4 +286,63 @@ void testReturnAllTags() {
         }
         assertThat(dtoList.get(0).getTagMessage(), is(not(tagList.get(2).getTagMessage())));
     }
+
+    @Test
+    @DisplayName("Test getGoalsByTag(JsonObject searchDto): returns list of GoalDto by Tag")
+    void testGetGoalsByTag() {
+        List<Tag> tags = new ArrayList<>();
+        Tag tag1 = new Tag();
+        tag1.setTagMessage("test1");
+        Tag tag2 = new Tag();
+        tag2.setTagMessage("test2");
+        Collections.addAll(tags, tag1);
+        List<Goal> goals = new ArrayList<>();
+        Goal goal1 = new Goal();
+        goal1.setUser(user1);
```

Comment:
I suggest adding a method that creates a `Goal` with the given parameters, or adding a constructor with parameters to `Goal` class.

Reply:
added the constructor with parameters to Goal class